woocommerce/woocommerce
# Prevent warning on wc_get_product_variation_attributes

### All Submissions:






### Changes proposed in this Pull Request:

Since [get_post_meta](https://github.com/woocommerce/woocommerce/blob/14a8b328a975a22b7d92602f471d27b06ade173d/plugins/woocommerce/includes/wc-product-functions.php#L689) can return `false` when the input is a boolean, zero or string, this PR handles this case to ensure that the return of get_post_meta is an array. This is important because the `$all_meta` is used in a [foreach](https://github.com/woocommerce/woocommerce/blob/14a8b328a975a22b7d92602f471d27b06ade173d/plugins/woocommerce/includes/wc-product-functions.php#L707) and prevents the warning `Invalid argument supplied for foreach()`.

### How to test the changes in this Pull Request:

1. Enable `debug.log` by adding the following to `wp-config.php`
```
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
define( 'WP_DEBUG_DISPLAY', false );
```
2. Run 
```
wp shell
wp> wc_get_product_variation_attributes( 0 )
```
3. You should not see a warning either in wp shell or in `debug.log`

### Other information:




### Changelog entry

> Dev - Prevent warning on wc_get_product_variation_attributes

### FOR PR REVIEWER ONLY:
